pkg/datapath/connector/config_test.go

```diff
@@ -0,0 +1,458 @@
+	return fakeAttr, nil
+}
+
+func destroyFakeLink(fakeAttr *fakeLinkAttributes) error {
+	if fakeAttr.WasCreated {
+		return link.DeleteByName(fakeAttr.Name)
+	}
+	return nil
+}
+
+func TestUseTunedBufferMargins(t *testing.T) {
+	tests := []struct {
+		name           string
+		datapathMode   string
+		expectedResult bool
+	}{
+		{
+			name:           "datapath-veth",
+			datapathMode:   datapathOption.DatapathModeVeth,
+			expectedResult: false,
+		},
+		{
+			name:           "datapath-netkit",
+			datapathMode:   datapathOption.DatapathModeNetkit,
+			expectedResult: true,
+		},
+		{
+			name:           "datapath-netkit-l2",
+			datapathMode:   datapathOption.DatapathModeNetkitL2,
+			expectedResult: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := useTunedBufferMargins(tt.datapathMode)
+			assert.Equal(t, tt.expectedResult, result)
+		})
+	}
+}
+
+func TestNewConfig(t *testing.T) {
+	logger := hivetest.Logger(t)
+
+	tests := []struct {
+		name           string
+		daemonConfig   *option.DaemonConfig
+		wgAgent        *fakeTypes.WireguardAgent
+		tunnelConfig   tunnel.Config
+		shouldError    bool
+		expectedConfig *ConnectorConfig
+	}{
+		{
+			name:           "datapath-veth",
+			daemonConfig:   &daemonConfigVeth,
+			wgAgent:        fakeTypes.NewTestAgent(wgConfigDisabled),
+			tunnelConfig:   tunnelConfigNative,
+			shouldError:    false,
+			expectedConfig: &ccTuningZero,
+		},
+		{
+			name:           "datapath-netkit",
+			daemonConfig:   &daemonConfigNetkit,
+			wgAgent:        fakeTypes.NewTestAgent(wgConfigDisabled),
+			tunnelConfig:   tunnelConfigNative,
+			shouldError:    false,
+			expectedConfig: &ccTuningZero,
+		},
+		{
+			name:           "datapath-netkit-l2",
+			daemonConfig:   &daemonConfigNetkitL2,
+			wgAgent:        fakeTypes.NewTestAgent(wgConfigDisabled),
+			tunnelConfig:   tunnelConfigNative,
+			shouldError:    false,
+			expectedConfig: &ccTuningZero,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			p := connectorParams{
+				Lifecycle:    &cell.DefaultLifecycle{},
+				Log:          logger,
+				Orchestrator: &fakeTypes.FakeOrchestrator{},
+				DaemonConfig: tt.daemonConfig,
+				WgAgent:      tt.wgAgent,
+				TunnelConfig: tt.tunnelConfig,
+			}
+			connector, err := newConfig(p)
+
+			switch tt.shouldError {
+			case true:
+				assert.Error(t, err)
+				assert.Nil(t, connector)
+
+			case false:
+				assert.NoError(t, err)
+				assert.NotNil(t, connector)
+				assert.Equal(t, tt.expectedConfig, connector)
+			}
+		})
+	}
+}
+
+func TestPrivilegedGenerateConfig(t *testing.T) {
+	testutils.PrivilegedTest(t)
+	logger := hivetest.Logger(t)
+
+	// Setup fake devices so we know the underlying magins of each driver
+	// on the system we're running on.
+	wgAttr, err := createFakeLink(fakeLinkWireguard)
+	if err != nil {
+		t.Fatalf("failed to create fake device %+v: %s", fakeLinkWireguard, err)
+	}
```

Comment:
```suggestion
	}
	defer destroyFakeLink(wgAttr)
```
Better deferring the execution right here. Calling it at the end of the function is a bit risky, because if something before fails (fatal, etc.) it will never be executed, and the host running the test would remain polluted with a fake interface.

Calls at the end of this function can be removed.